tests/sql/test_db_connector.py

```diff
@@ -0,0 +1,138 @@
+"""
+Unit tests for DbConnector class.
+"""
+#standard imports
+from json import JSONEncoder
+
+# third party imports
+
+# project imports
+from lib.sql.db_connector import DbConnector, Tag, File, Hash
+
+def test_insert_tag(dbsession):
+    """
+    Unit tests for insert_tag method.
+    """
+
+    techno = "jQuery"
+    tag = "1.2.3"
+
+    DbConnector.insert_tag(dbsession, techno, tag)
+
+    inserted_tag = dbsession.query(Tag).first()
+    assert inserted_tag.technology == techno
+    assert inserted_tag.tag == tag
+
+    DbConnector.insert_tag(dbsession, techno, tag)
+    # tag already added before
+    assert dbsession.query(Tag).count() == 1
+
+
+def test_insert_tags(dbsession):
+    """
+    Unit tests for insert_tags method.
+    """
+
+    techno = "jQuery"
+    tags = ["1.2.3", "1.3.4", "1.3.5"]
+
+    DbConnector().insert_tags(dbsession, techno, tags)
+
+    inserted_tags = dbsession.query(Tag)
+    assert inserted_tags.count() == 3
+    assert inserted_tags[0].technology == techno
+    assert inserted_tags[0].tag == tags[0]
+    assert inserted_tags[1].technology == techno
+    assert inserted_tags[1].tag == tags[1]
+    assert inserted_tags[2].technology == techno
+    assert inserted_tags[2].tag == tags[2]
+
+
+def test_get_tags(dbsession):
+    """
+    Unit tests for get_tags method.
+    """
+
+    techno = "jQuery"
+    tags = ["1.2.3", "1.3.4"]
+
+    dbsession.add(Tag(technology=techno, tag=tags[0]))
+    dbsession.add(Tag(technology=techno, tag=tags[1]))
+
+    retrieved_tags = DbConnector.get_tags(dbsession, techno)
+    assert len(retrieved_tags) == 2
+    assert retrieved_tags[0].technology == techno
+    assert retrieved_tags[0].tag == tags[0]
+    assert retrieved_tags[1].technology == techno
+    assert retrieved_tags[1].tag == tags[1]
+
+
+def test_insert_file(dbsession):
+    """
+    Unit tests for insert_file method.
+    """
+
+    techno = "jQuery"
+    path = "/src/jquery.min.js"
+
+    DbConnector.insert_file(dbsession, techno, path)
+
+    inserted_file = dbsession.query(File).first()
+    assert inserted_file.technology == techno
+    assert inserted_file.path == path
+
+    DbConnector.insert_file(dbsession, techno, path)
+    # file already added before
+    assert dbsession.query(File).count() == 1
+
+
+def test_insert_or_update_hash(dbsession):
+    """
+    Unit tests for insert_or_update_hash method.
+    """
+
+    hash_value = "ab78a59a6f2fbacdceb42bc4b6f1aeff0445d119a83128a20786d34bc3f64527"
+    techno = "jQuery"
+    versions = ["1.2.3", "1.3.4"]
+
+    DbConnector.insert_or_update_hash(dbsession, hash_value, techno, versions[0])
+
+    inserted_hash = dbsession.query(Hash).first()
```

Comment:
Here it would be clearer if you could refactor this into 3 separated tests with explicit names I think